src/errors.rs

```diff
@@ -77,6 +93,8 @@ impl fmt::Display for Error {
             Error::InvalidResponse(ref cause) => {
                 write!(f, "Response doesn't have the expected format: {}", cause)
             }
+            Error::Ssl(ref err) => err.fmt(f),
+            Error::Mime(ref err) => err.fmt(f),
```

Comment:
Instead of printing the source error, we could also return it in `StdError::source` and either print nothing here or a message "on top" of what happened.

Also see: https://users.rust-lang.org/t/do-people-not-care-about-printable-error-chains-a-k-a-how-to-nicely-implement-display-for-an-error/35362/2

Reply:
Made two new commits. Not really sure if second commit was necessary, please check it and i will undo it if needed. The problem with it is that you said that i could either "print nothing here" (Which is definitely not an option) or print a message "on top" of what happened, which will output something like this with Mime type: ```Mime error: mime parse error: <ParseError>``` (See how mime::FromStrError displays itself at its [source code](https://docs.rs/mime/0.3.16/src/mime/lib.rs.html#82-86)). As you can see it does not look really good, first two statements just repeat each other. I would remove this "Mime error:" at the beginning and leave just ```err.fmt(f)``` in code how it was before. If i'm wrong and i undo these changes, i might need to redo them again, so just wanna see what you think about that before doing anything

Reply:
If we don't have anything more meaningful to print on top of the original error, why is printing nothing not an option? Together with returning the `source`, libraries like `anyhow` will print the entire chain of errors and thus print the "inner" error.

But, I think the better option would be to move away from a generall "Ssl" error and be more specific of what failed. I commented inline with some ideas, feel free to ignore if you find them too verbose :)